src/azure-cli/azure/cli/command_modules/acs/azuremonitormetrics/recordingrules/create.py

```diff
@@ -50,23 +51,30 @@ def put_rules(cmd, default_rule_group_id, default_rule_group_name, mac_region, a
 
 # pylint: disable=line-too-long
 def create_rules(cmd, cluster_subscription, cluster_resource_group_name, cluster_name, azure_monitor_workspace_resource_id, mac_region, raw_parameters):
+    # limit rule group name to 260 characters
     # with urllib.request.urlopen("https://defaultrulessc.blob.core.windows.net/defaultrules/ManagedPrometheusDefaultRecordingRules.json") as url:
     #     default_rules_template = json.loads(url.read().decode())
     default_rules_template = get_recording_rules_template(cmd, azure_monitor_workspace_resource_id)
-    default_rule_group_name = "NodeRecordingRulesRuleGroup-{0}".format(cluster_name)
+    default_rule_group_name = "NodeRecordingRulesRuleGroup-{0}".format(cluster_name[:232])
     default_rule_group_id = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.AlertsManagement/prometheusRuleGroups/{2}".format(
         cluster_subscription,
         cluster_resource_group_name,
         default_rule_group_name
     )
+    cluster_resource_id = \
+        "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.ContainerService/managedClusters/{2}".format(
+            cluster_subscription,
+            cluster_resource_group_name,
+            cluster_name
+        )
     url = "{0}{1}?api-version={2}".format(
         cmd.cli_ctx.cloud.endpoints.resource_manager,
         default_rule_group_id,
         RULES_API
     )
-    put_rules(cmd, default_rule_group_id, default_rule_group_name, mac_region, azure_monitor_workspace_resource_id, cluster_name, default_rules_template, url, True, 0)
+    put_rules(cmd, default_rule_group_id, default_rule_group_name, mac_region, cluster_resource_id, azure_monitor_workspace_resource_id, cluster_name, default_rules_template, url, True, 0)
 
-    default_rule_group_name = "KubernetesRecordingRulesRuleGroup-{0}".format(cluster_name)
+    default_rule_group_name = "KubernetesRecordingRulesRuleGroup-{0}".format(cluster_name[:226])
```

Comment:
use a helper function for truncation?

Reply:
https://github.com/Azure/azure-cli/pull/27788/files